lib/sync/api-syncRecords.js

```diff
@@ -183,8 +183,9 @@ function syncRecords(datasetId, params, cb) {
           return callback(err);
         }
         if (!datasetClientJson) {
```

Comment:
This is the part that I mentioned in the commit message, where it returns a different type as the first parameter to the callback in different situations.